languages/swift/exercises/concept/closures/.docs/instructions.md

```diff
@@ -0,0 +1,111 @@
+
+```swift
+let powersOf3 = unroll(1,
+                       generator: { (n: Int) -> (seed: Int, next: Int) in (seed: 3 * n, next: n) },
+                       stop: { (n: Int) -> Bool in n > 1_000_000 })
+// => [1, 3, 9, 27, 81, 243, 729, 2187, 6561, 19683, 59049, 177147, 531441]
+```
+
+You will use unroll to help you generate a few sequences.
+
+## 1. Write helper closures to generate a sequence of digits
+
+One can generate a sequence of digits of a number, _n_ by computing `n % 10` to get the last digit of `n` and then dividing `n` by 10 to remove that last digit. This process can then be repeated until `n` is 0.
+
+For this task, you need to define a closure of type `(Int) -> (seed: Int, next: Int)` That takes a number and returns that number with its last digit removed as `seed:` along with the last digit as `next:`. Assign this closure to a constant named `generateDigit`.
+
+You will also need to generate a closure of type `(Int) -> Bool` That returns `true` if n = 0 and `false` otherwise. Assign this closure to a constant named `stopAtZero`.
+
+```swift
+generateDigit(123)
+// => (seed: 12, next: 3)
+
+stopAtZero(0)
+// => true
+```
+
+## 2. Generate a sequence of digits of an integer
+
+Use these helper closures with unroll to define a closure of type `(Int) -> [Int]` that takes an Int as input and returns the sequence of digits. Assign this closure to a constant named `digitsOf`.
+
+```swift
+digitsOf(123)
+// => [3, 2, 1]
+```
+
+## 3. Write helper closures to generate a sequence of bits
+
+One can generate a sequence of digits of a number, _n_ by computing `n % 2` to get the last bit of `n` and then dividing `n` by 2 to remove that last bit. This process can then be repeated until `n` is 0.
+
+For this task, you need to define a closure of type `(Int) -> (seed: Int, next: Int)` That takes a number and returns that number with its last digit removed as `seed:` along with the last digit as `next:`. Assign this closure to a constant named `generateBit`.
+
+```swift
+generateBit(11)
+// => (seed: 5, next: 1)
+```
+
+## 4. Generate a sequence of bits of an integer
+
+Use these helper closures with unroll to define a closure of type `(Int) -> [Int]` that takes an Int as input and returns the sequence of bits. Assign this closure to a constant named `bitsOf`.
+
+```swift
+bitsOf(11)
+// => [1, 1, 0, 1]
+```
+
+## 5. Generate a closure for any base from 2-10
+
+Implement the function:
+
+```swift
+generatorForBase(_ base: Int) -> ((Int) -> (seed: Int, next: Int))?
+```
+
+That takes an integer, `base` and returns an optional closure. The function should return a closure that works just like `generateDigit` or `generateBit`, but for base `base` rather than just base 2 or 10.
+
+```swift
+let gen7 = generatorForBase(7)
+// => (Int) -> (seed: Int, next: Int)
+gen7!(123)
+// => (seed 17, next 4)
+```
+
+## 6. Generate a Collatz sequence from a positive integer
```

Comment:
The `collatz-conjecture` is an existing v2 exercise (see https://github.com/exercism/swift/blob/master/exercises/collatz-conjecture/README.md) and thus ineligible for usage in a Concept Exercise.